plugins/state_history_plugin/tests/session_test.cpp

```diff
@@ -80,9 +80,13 @@ fc::datastream<ST>& operator>>(fc::datastream<ST>& ds, eosio::state_history::get
 
 //------------------------------------------------------------------------------
 
-fc::sha256 block_id_for(const uint32_t bnum) {
-   fc::sha256 m = fc::sha256::hash(fc::sha256::hash(std::to_string(bnum)));
+std::unordered_map<uint32_t, eosio::chain::block_id_type> block_ids;
+fc::sha256 block_id_for(const uint32_t bnum, const std::string& nonce = {}) {
+   if (block_ids.count(bnum))
+      return block_ids[bnum];
```

Comment:
Small inefficiency. This (or using .contains()) will do 2 lookups of the ley in the unordered_map. It would be more optimal to do:
```
if (auto it = block_ids.find(bnum); it != block_ids.end())
    return it->second;
```

Reply:
Sure, it is test code and the second lookup should be fast as it just found it.

Reply:
Indeed. It seems I can't help myself be a stickler for details :-).
The second lookup will be faster only because the memory is in the processor cache, otherwise, if there are collisions, it may still iterate over a linked list and do comparisons.